exist-core/src/main/java/org/exist/xquery/ForExpr.java

```diff
@@ -229,7 +229,11 @@ private void processItem(LocalVariable var, Item contextItem, Sequence in, Seque
         context.proceed(this);
         context.setContextSequencePosition(p, in);
         if (positionalVariable != null) {
-            at.setValue(new IntegerValue(this, p + 1));
+            if (contextItem.getType() == Type.ANY_ATOMIC_TYPE) {
```

Comment:
how about
```
final int position = contextItem.getType() == Type.ANY_ATOMIC_TYPE ? 0 : p + 1;
at.setValue(new IntegerValue(position));
```

Reply:
I am also wondering, if there is a better way to check for this case.
Since we want to avoid that the position is set to 0 in any other case.
Since the function `processItem` is called with contextItem set to `AtomicValue.EMPTY_VALUE` if and only if `allowing empty` is set _and_ there are no items in the binding sequence we should best check for that.
What do you think?

Reply:
Yeah that is better
Will make that change